### DISCUSSION TITLE
VR Mouse Cursor Auto Disable

### DISCUSSION THREAD

**Discussion by cajunjosh** [magknight/787aviatorsTracker#712] [Ideas]
Current Behavior:

With custom cursor enabled VR users can use the toggle vr cursor bind to disable the 3D cursor and then manipulate touch screen inputs using the Magknight magenta custom cursor.  

Once you have clicked in a touch screen window it keyboard locks you so that you can enter details such as DEPT and DEST in a flight plan using keyboard inputs.  

The only way to exit this entry and release the keyboard capture is to look away from the touch screen with the VR mouse cursor disabled and click an empty area.   We can then toggle the VR mouse cursor back on and continue as usual.  

Desired Behavior:

User must enable custom cursor in settings.
When it is detected that the 3D VR mouse cursor has entered the perimeter of a touch screen interface it will automatically toggle off leaving only the custom cursor to manipulate entries.
Allow the standard keyboard capture if any data field has been clicked but automatically release the keyboard capture if either the enter key is pressed on the keyboard or the mouse cursor is moved off the touch screen interface and clicked outside the touch screen.
When it is detected that the mouse cursor has moved outside of the perimeter of a touch screen interface the 3D VR mouse cursor is automatically re-enabled.

A picture is worth a 1000 words so I recorded a video of the Magknight and the Flight Factor A350 showing current and desired behavior.  Thanks in advance for any consideration given to this update!

https://www.youtube.com/watch?v=Qk8u2vXpgno

**Comment by aeroniemi** [comment]
So, in 1.7.24 I've changed the Mfd keyboard entry to work globally rather than by panel - this means that you don't have to click on or off of a screen for keyboard entry, just need to have the mouse over the screen. This i think fixes the keyboard thing, but is probably less desirable for non-vr users as it'll make life a bit more difficult, but we'll see what people say


My understanding of the other bit is that you want the VR cursor to turn off when you enter a screen, and on when you leave it - that seems to be somewhat more difficult than it sounds as I don't think we get any kind of info about where the mouse is with the vr cursor on

**Reply by cajunjosh** [reply]
Tha k you for checking into it Aero.  Wasn't sure if it was a simple solution that Flight factor had done or something more complicated but definitely appreciate your time looiking over it.  Look forward to testing the new changes to .24. Hopefully XP12 improves some of the VR cursor behaviors to take the work load off the devs.
